ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/AbstractSchemaDefinitions.java

```diff
@@ -28,19 +32,27 @@ public abstract class AbstractSchemaDefinitions implements SchemaDefinitions {
   final SszBitvectorSchema<SszBitvector> syncnetsENRFieldSchema =
       SszBitvectorSchema.create(NetworkConstants.SYNC_COMMITTEE_SUBNET_COUNT);
   private final HistoricalBatchSchema historicalBatchSchema;
-  private final IndexedAttestation.IndexedAttestationSchema indexedAttestationSchema;
-  private final AttesterSlashing.AttesterSlashingSchema attesterSlashingSchema;
+  private final IndexedAttestationSchema<?> indexedAttestationSchema;
+  private final AttesterSlashingSchema<?> attesterSlashingSchema;
   private final BeaconBlocksByRootRequestMessage.BeaconBlocksByRootRequestMessageSchema
       beaconBlocksByRootRequestMessageSchema;
 
   public AbstractSchemaDefinitions(final SpecConfig specConfig) {
     this.historicalBatchSchema = new HistoricalBatchSchema(specConfig.getSlotsPerHistoricalRoot());
-    this.indexedAttestationSchema =
-        new IndexedAttestation.IndexedAttestationSchema(
-            getMaxValidatorPerAttestation(specConfig), toVersionElectra().isPresent());
-    this.attesterSlashingSchema =
-        new AttesterSlashing.AttesterSlashingSchema(
-            indexedAttestationSchema, toVersionElectra().isPresent());
+    if (toVersionElectra().isPresent()) {
```

Comment:
Not a big fan of this and I'm wondering if we can get rid of it since we're on it. Isn't it possible to remove the `getIndexedAttestationSchema` and `getAttesterSlashingSchema` from the `AbstractSchemaDefinitions` and rather override them directly in `SchemaDefinitionsPhase0` and `SchemaDefinitionsElectra` the same way we did for the `getAttestationSchema`?

Reply:
I haven't even tried cose I assumed some dependency issues. Going to try!

Reply:
done! nice catch!